packagePrefix is null

I upgraded to quarkus 3.29.0 and also tried to upgrade vaadin (have been on 24.5.8)

Starting with Vaadin and Vaadin-Quarkus 24.6.6 (up to and including 24.6.5 it works well) (i tried so many versions to find the exact version where this starts) i get the following error when trying to execute quarkusTest classes.

`java.lang.NullPointerException: Cannot invoke "String.startsWith(String)" because the return value of "org.jboss.jandex.DotName.packagePrefix()" is null`

I tried all possible jandex voodoo that ai was able to come up with. However this dis not seem to help.

with 24.6.5. i also get errors from rollup when building the frontend bundle.

What am i missing?

I don't really get how the versions in this description of the project (2.2., 2.1 ...) relate to the versions in the maven repository which have the same version numbers as vaadin. So how can i find out which versions of vaadin, quarkus, and vaadin-quarkus are compatible?

It's not the first time that i am struggeling to find a working combination of the three. I'm currently stuck on 24.5.8

To have the most usable stacktrace i used 24.9.0 and tried to build with the following exception. This Exception is raised for every Test class which is annotated with @QuarkusTest. Including very simple test cases as well as integration tests for database integrations.

Another update: The rollup errors i was talking about are suddenly gone. Probably because i restartet everything. That means: i can run everything perfectly fine, i just can't execute the tests anymore.

```java
org.gradle.api.internal.tasks.testing.TestSuiteExecutionException: Could not execute test class 'com.example.starter.base.core.adapter.eclipsestore.DocumentStoreTest'.
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:55)
	at java.base@17.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@17.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base@17.0.12/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@17.0.12/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:92)
	at jdk.proxy1/jdk.proxy1.$Proxy4.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:183)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:132)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:103)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:63)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:122)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:72)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor#vaadinNativeSupport threw an exception: java.lang.NullPointerException: Cannot invoke "String.startsWith(String)" because the return value of "org.jboss.jandex.DotName.packagePrefix()" is null
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.lambda$collectClassesInPackage$11(VaadinQuarkusNativeProcessor.java:352)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.getCommonComponentClasses(VaadinQuarkusNativeProcessor.java:388)
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.vaadinNativeSupport(VaadinQuarkusNativeProcessor.java:298)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:840)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:375)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:292)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)
	at io.quarkus.test.junit.AppMakerHelper.getStartupAction(AppMakerHelper.java:253)
	at io.quarkus.test.junit.classloading.FacadeClassLoader.getOrCreateRuntimeClassLoader(FacadeClassLoader.java:587)
	at io.quarkus.test.junit.classloading.FacadeClassLoader.getQuarkusClassLoader(FacadeClassLoader.java:484)
	at io.quarkus.test.junit.classloading.FacadeClassLoader.loadClass(FacadeClassLoader.java:366)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.loadClass(JUnitPlatformTestClassProcessor.java:173)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.access$100(JUnitPlatformTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.execute(JUnitPlatformTestClassProcessor.java:109)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.execute(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:54)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:53)
	... 18 more
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor#vaadinNativeSupport threw an exception: java.lang.NullPointerException: Cannot invoke "String.startsWith(String)" because the return value of "org.jboss.jandex.DotName.packagePrefix()" is null
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.lambda$collectClassesInPackage$11(VaadinQuarkusNativeProcessor.java:352)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.getCommonComponentClasses(VaadinQuarkusNativeProcessor.java:388)
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.vaadinNativeSupport(VaadinQuarkusNativeProcessor.java:298)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:840)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:139)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:78)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:371)
	... 32 more
Caused by: java.lang.NullPointerException: Cannot invoke "String.startsWith(String)" because the return value of "org.jboss.jandex.DotName.packagePrefix()" is null
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.lambda$collectClassesInPackage$11(VaadinQuarkusNativeProcessor.java:352)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.getCommonComponentClasses(VaadinQuarkusNativeProcessor.java:388)
	at com.vaadin.quarkus.deployment.VaadinQuarkusNativeProcessor.vaadinNativeSupport(VaadinQuarkusNativeProcessor.java:298)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:872)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at java.base/java.lang.Thread.run(Thread.java:840)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

--- a/deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusNativeProcessor.java
+++ b/deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusNativeProcessor.java
@@ -252,7 +252,7 @@ public class VaadinQuarkusNativeProcessor {
         }
     }
 
-    @BuildStep
+    @BuildStep(onlyIf = IsNativeBuild.class)
     void vaadinNativeSupport(CombinedIndexBuildItem combinedIndex,
             BuildProducer<RuntimeInitializedPackageBuildItem> runtimeInitializedPackage,
             BuildProducer<NativeImageResourcePatternsBuildItem> nativeImageResource,
@@ -350,10 +350,11 @@ public class VaadinQuarkusNativeProcessor {
     private Stream<ClassInfo> collectClassesInPackage(IndexView index,
             String basePackage, boolean recursive) {
         Predicate<ClassInfo> predicate = recursive
-                ? classInfo -> classInfo.name().packagePrefix()
-                        .startsWith(basePackage)
-                : classInfo -> classInfo.name().packagePrefix()
-                        .equals(basePackage);
+                ? classInfo -> classInfo.name().packagePrefix() != null
+                        && classInfo.name().packagePrefix()
+                                .startsWith(basePackage)
+                : classInfo -> classInfo.name().packagePrefix() != null
+                        && classInfo.name().packagePrefix().equals(basePackage);
         return index.getKnownClasses().stream().filter(predicate);
     }
 

--- a/deployment/src/test/java/ClassInDefaultPackage.java
+++ b/deployment/src/test/java/ClassInDefaultPackage.java
@@ -1,0 +1,18 @@
+/*
+ *  Copyright 2000-2025 Vaadin Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+public class ClassInDefaultPackage {
+}

--- a/deployment/src/test/java/com/vaadin/flow/quarkus/test/NativeBuildClassesInDefaultPackageTest.java
+++ b/deployment/src/test/java/com/vaadin/flow/quarkus/test/NativeBuildClassesInDefaultPackageTest.java
@@ -1,0 +1,47 @@
+/*
+ *  Copyright 2000-2025 Vaadin Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+package com.vaadin.flow.quarkus.test;
+
+import io.quarkus.test.QuarkusUnitTest;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class NativeBuildClassesInDefaultPackageTest {
+
+    // Start unit test with your extension loaded
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(loadClassFromDefaultPackage()))
+            .overrideConfigKey("quarkus.package.type", "native");
+
+    @Test
+    public void classesInDefaultPackage_vaadinNativeSupport_scanShouldNotFail() {
+        Assertions.assertTrue(true,
+                "Quarkus build should not fail with classes in default package");
+    }
+
+    private static Class<?> loadClassFromDefaultPackage() {
+        try {
+            return Class.forName("ClassInDefaultPackage");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}

